libs/wampcc/wamp_router.cc

```diff
@@ -264,12 +264,14 @@ std::future<uverr> wamp_router::listen(auth_provider auth,
                               json_object& details,
                               wamp_args& args)
     {
+      ws.authorize(uri, auth_provider::action::call);
       this->handle_inbound_call(&ws, reqid, uri, details, args);
     };
 
     handlers.on_publish =
     [this](wamp_session& ws, t_request_id request_id, std::string uri,
            json_object options, wamp_args args) {
+      ws.authorize(uri, auth_provider::action::publish);
```

Comment:
wonder if the authorize() check on line 274 should be inside the try/catch ?  Suppose it depends on what is desired; should the router drop the client, or, just return error reply to this single message?  Can review after merge.

Reply:
This way if the call fails due to authorization the caller get's the proper error response.